src/observed.decorator.spec.ts

```diff
@@ -6,21 +6,61 @@ const testValue1 = 10;
 const testValue2 = 25;
 const testValue3 = 50;
 
-class TestClass {
+interface Test {
+    property: any;
+    property$: Observable<any>;
+}
+
+class TestClass implements Test {
```

Comment:
If we are going to use this interface, we should do it with a generic type as follows. Also, it seems that `implements Test` is missing from many of the new test classes.

```suggestion
interface Test<T> {
    property: T;
    property$: Observable<T>;
}

class TestClass implements Test<number> {
```

Reply:
Considering the templating code suggestion seemingly isn't valid typescript and, more importantly, that **this is a spec file**, I won't be making any changes here.

The interface was only used so that I could re-use the `checkObservable` function, which is only used for instantiated classes with the default `@Observed()` functionality (i.e. `TestClass` & `TestClassNotInitialized`).

All of the classes with static properties cannot implement this `Test` interface because they have identical static properties. Implementing the interface expects the properties to be non-static

The only additional class that might warrant implementing the `Test` interface is `TestSubject`, which wouldn't actually provide any benefit, seeing as it's only used in one place with no generics, so I'm leaving it as is.